src/main/java/org/opensearch/neuralsearch/processor/normalization/MinMaxScoreNormalizationTechnique.java

```diff
@@ -167,22 +224,197 @@ private float[] getMinScores(final List<CompoundTopDocs> queryTopDocs, final int
-        return normalizedScore == 0.0f ? MIN_SCORE : normalizedScore;
+        if (lowerBound.isEnabled() == false) {
+            return LowerBound.Mode.IGNORE.normalize(score, minScore, maxScore, lowerBound.getMinScore());
+        }
+        return lowerBound.getMode().normalize(score, minScore, maxScore, lowerBound.getMinScore());
+    }
+
+    /**
+     * Get lower bounds from input parameters
+     * @param params user provided input parameters for this technique
+     * @return optional list of lower bounds. Can be empty in case lower bounds are not provided
+     */
+    private Optional<List<Pair<LowerBound.Mode, Float>>> getLowerBounds(final Map<String, Object> params) {
+        // validate that the input parameters are in correct format
+        if (Objects.isNull(params) || params.containsKey(PARAM_NAME_LOWER_BOUNDS) == false) {
+            return Optional.empty();
+        }
+
+        Object lowerBoundsObj = params.get(PARAM_NAME_LOWER_BOUNDS);
+        if (lowerBoundsObj instanceof List<?> == false) {
+            throw new IllegalArgumentException("lower_bounds must be a List");
+        }
+        List<?> lowerBoundsParams = (List<?>) lowerBoundsObj;
+        // number of lower bounds must match the number of sub-queries in a hybrid query
+        if (lowerBoundsParams.size() > MAX_NUMBER_OF_SUB_QUERIES) {
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.ROOT,
+                    "lower_bounds size %d should be less than or equal to %d",
+                    lowerBoundsParams.size(),
+                    MAX_NUMBER_OF_SUB_QUERIES
+                )
+            );
+        }
+        // parse each lower bound item and put all items in a list
+        List<Pair<LowerBound.Mode, Float>> lowerBounds = lowerBoundsParams.stream().map(this::parseLowerBound).collect(Collectors.toList());
+
+        return Optional.of(lowerBounds);
+    }
+
+    @SuppressWarnings("unchecked")
+    /**
+     * Parse each lower bound item and return a pair of mode and min score
+     * @param boundObj lower bound item provided by the client
+     * @return a single pair of mode and min score
+     */
+    private Pair<LowerBound.Mode, Float> parseLowerBound(Object boundObj) {
+        if ((boundObj instanceof Map) == false) {
+            throw new IllegalArgumentException("each lower bound must be a map");
+        }
+
+        Map<String, Object> lowerBound = (Map<String, Object>) boundObj;
+
+        String lowerBoundModeValue = Objects.toString(lowerBound.get(PARAM_NAME_LOWER_BOUND_MODE), "");
+        LowerBound.Mode mode = LowerBound.Mode.fromString(lowerBoundModeValue);
+        float minScore = extractAndValidateMinScore(lowerBound);
+
+        return ImmutablePair.of(mode, minScore);
+    }
+
+    private float extractAndValidateMinScore(Map<String, Object> lowerBound) {
+        Object minScoreObj = lowerBound.get(PARAM_NAME_LOWER_BOUND_MIN_SCORE);
+        if (minScoreObj == null) {
```

Comment:
A question here. If a user is defining lower bounds then shouldn't we mandate them to provide a min score otherwise is there any point to define lower bound?

Reply:
I think that was one of the asks when we reviewed the design/RFC, we can use default of 0.0 that is giving reasonable results and simplify the interface for end user

Reply:
not sure what do you mean. I was referring to scenario when user sets just lower bound, in this case we use 0.0 score as min_score for that lower bound. Reason for it is a simpler request syntax and relatively good defaults for entry level users. User can be not aware of optimal min_score for lower bound, or can be fine with our defaults.

Reply:
So in the scenario where user just define a lower bound but doesn't pass any min score then it will be directed to default value, right?
Wouldn't in such scenario rather than using the default min value we should have used the original min score we get from the sub query itself?